backend_kvm.go

```diff
@@ -0,0 +1,136 @@
+// +build linux
+// +build amd64
+
+package fakemachine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type KvmBackend struct {}
+
+func init() {
+	RegisterBackend(KvmBackend{})
+}
+
+func (b KvmBackend) Name() string {
+	return "kvm"
+}
+
+func (b KvmBackend) Supported() (bool, error) {
+	if _, err := os.Stat("/dev/kvm"); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (b KvmBackend) InitScriptModules() []string {
+	return []string{"virtio_pci", "virtio_console", "9pnet_virtio", "9p"}
+}
+
+func (b KvmBackend) InitScriptExtraCommands() []string {
+	return nil
+}
+
+func (b KvmBackend) NetworkdMatch() string {
+	return "e*"
+}
+
+func (b KvmBackend) CmdMountVolume(mount mountPoint) string {
+	return fmt.Sprintf("busybox mount -v -t 9p -o trans=virtio,version=9p2000.L,cache=loose,msize=262144 %s %s", mount.label, mount.machineDirectory)
+}
+
+func (b KvmBackend) GetFstabEntry(mount mountPoint) string {
+	return fmt.Sprintf("%s %s 9p trans=virtio,version=9p2000.L,cache=loose,msize=262144 0 0", mount.label, mount.machineDirectory)
+}
+
+func (b KvmBackend) GetImagePath(hostPath string, label string, index int) string {
+	return fmt.Sprintf("/dev/disk/by-id/virtio-%s", label)
+}
+
+func (b KvmBackend) GetJobOutputTTY(m *Machine) string {
```

Comment:
The one thing i'm wary about here is that we're exposing the full Machine object to the backend which makes the contract a bit vague. As it exposes the backend to a lot of implementation details (what properties fo the Machine should a backend take into account?). 

So it would probably be nicer to just provide a showBoot bool